benchmarks/src/main/java/io/druid/benchmark/QueryableIndexLoadingBenchmark.java

```diff
@@ -0,0 +1,293 @@
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Thread)
+public class QueryableIndexLoadingBenchmark
+{
+  private class IndexedIntsDimensionSelectorRow implements Iterable<Comparable>
+  {
+    private final IndexedInts delegate;
+
+    public IndexedIntsDimensionSelectorRow(IndexedInts delegate)
+    {
+      this.delegate = delegate;
+    }
+
+    public Iterator iterator()
+    {
+      return delegate.iterator();
+    }
+
+    public int size()
+    {
+      return delegate.size();
+    }
+
+    public Comparable get(int index)
+    {
+      return delegate.get(index);
+    }
+
+    public IndexedInts getDelegate()
+    {
+      return delegate;
+    }
+  }
+
+  static final int ROW_SIZE = 64;
+  static final int MAX_ROWS = 250000;
+  static final int MAX_VALUE = 20000;
+
+  private static final IndexMergerV9 INDEX_MERGER_V9;
+  private static final IndexIO INDEX_IO;
+  public static final ObjectMapper JSON_MAPPER;
+
+  private QueryableIndex qIndex;
+
+  private static AggregatorFactory[] AGGS;
+
+  private Random rng;
+  private Random rng2;
+  private File tmpFile;
+
+  private StorageAdapter qAdapter;
+
+  private boolean initialized = false;
+
+  static {
+    JSON_MAPPER = new DefaultObjectMapper();
+    INDEX_IO = new IndexIO(
+        JSON_MAPPER,
+        new ColumnConfig()
+        {
+          @Override
+          public int columnCacheSizeBytes()
+          {
+            return 0;
+          }
+        }
+    );
+    INDEX_MERGER_V9 = new IndexMergerV9(JSON_MAPPER, INDEX_IO);
+  }
+
+  static {
+    final ArrayList<AggregatorFactory> ingestAggregatorFactories = new ArrayList<>(ROW_SIZE + 1);
+    ingestAggregatorFactories.add(new CountAggregatorFactory("rows"));
+    for (int i = 0; i < ROW_SIZE; ++i) {
+      if (i % 2 != 0) {
+        continue;
+      }
+
+      ingestAggregatorFactories.add(
+          new LongSumAggregatorFactory(
+              String.format("sumResult%s", i),
+              String.format("Dim_%s", i)
+          )
+      );
+      ingestAggregatorFactories.add(
+          new DoubleSumAggregatorFactory(
+              String.format("doubleSumResult%s", i),
+              String.format("Dim_%s", i)
+          )
+      );
+    }
+    AGGS = ingestAggregatorFactories.toArray(new AggregatorFactory[0]);
+  }
+
+  @Setup
+  public void setup() throws IOException
+  {
+    if(initialized) {
+      return;
+    }
+
+    rng = new Random(9999);
+    rng2 = new Random(9999);
+
+
+    /*
```

Comment:
@fjy 

It's sort of rough, but I had two "modes" of behavior at that area in the benchmark, one was to generate a fresh QueryableIndex, and the other was to load a pre-generated index from disk.

I've uncommented out that block which handles the former case, left a commented line and comment about loading from a file